### ISSUE TITLE
Issue/request: icon in link

### ISSUE THREAD

**Issue by madeinnature** [quellenform/t3x-iconpack#14]
Hello,
THX for the great extension!
I would like to have the ability to use icons in links.
If I now link a text that contains an icon, the following happens:
`<a href="t3://page?uid=1">Sed cursus turpis </a><span class="fa-solid fa-arrow-down-to-square" data-iconfig="fa6p:solid,arrow-down-to-square"></span><a href="t3://page?uid=1">vitae tortor</a>`
Of course, it would be desirable for the icon to be included within the a-tag.
Even if it is not particularly user-friendly, I would also appreciate the ability to link an icon by itself.

**Comment by DirkPersky** [comment]
He,

This issue is becouse to the `span` tag.

I tested your extension today and it looks interesting.
If you woud develop your extension further the next LTS Version.
I maybe would deprecated my https://extensions.typo3.org/extension/rte_ckeditor_fontawesome/ for it.

**Comment by stephankellermayr** [comment]
@DirkPersky 

Yes, you could actually be right: The behaviour is somehow related to the `span` tag. However, I haven't yet found out exactly how to fix it.

And yes, I will definitely continue to develop this extension. However, I would be happy if other capable developers could participate in this project, because it is a relatively high effort alone, and as you know, there is little to no budget for open source projects.

In any case, I welcome your interest in the extension and would be happy if we could work together here, especially since your extension was also a certain inspiration for me (or rather its lack of functionality).

**Comment by DirkPersky** [comment]
@stephankellermayr

your welcome.

If u change back the `span` to the `i` Tag it works.
But im sure you had a reason to use the `span` Tag.

I created my extension on the reasone there was no alternative in the store and be happy there will be an alternative 👍

**Comment by stephankellermayr** [comment]
@DirkPersky Strictly speaking, the use of the `i` tag is semantically incorrect and the `span` tag should always be used for icons.
Nevertheless, it does not really solve the underlying problem in depth, because there are also iconpacks (e.g. [iconpack_typo3](https://github.com/quellenform/t3x-iconpack-typo3)) that come completely without a webfont and only use SVG graphics. In this case, an `img` tag is used in the CKEDitor5, which does not allow embedding in a link, just like the `i` tag.

It is therefore actually a problem of the [TYPO3 link plugin](https://github.com/TYPO3/typo3/blob/main/Build/Sources/TypeScript/rte_ckeditor/plugin/typo3-link.ts), which treats all of these elements as independent elements and excludes them from the link.

I'm afraid I won't be able to solve the problem on the part of Iconpack, because the bug should actually be fixed in the link plugin.

**Comment by ftrojahn** [comment]
Is there any news on this?

As workaround we found to be working: putting an icon within some text, set a link around and then delete the text before and after the icon.

```
<p><a class="external-link-new-window" href="https://www.domain.tld" target="_blank" title="Link öffnen">Link-Text</a></p>

<p><span class="fa-x-twitter fab" data-iconfig="fa6:brands,x-twitter"></span><a class="external-link-new-window" href="https://www.domain.tld" target="_blank" title="Link öffnen">Link-Text mit Icon vorn</a></p>

<p><a class="internal-link" href="t3://page?uid=238378">Link-Text&nbsp;<span class="fa-anchor-circle-check fas" data-iconfig="fa6:solid,anchor-circle-check"></span> Icon in der Mitte</a></p>

<p><a class="external-link-new-window" href="https://www.domain.tld" target="_blank" title="Link öffnen"><span class="fa-x-twitter fab" data-iconfig="fa6:brands,x-twitter"></span></a></p>
```

But this isn't so nice to explain to editors.

Does a forge issue already exist for that, or, is there an idea, how to patch/fix the `typo3-link.ts`?

**Comment by stephankellermayr** [comment]
@ftrojahn Hi and thanks for your feedback on this topic! As far as I know, there is no forge issue yet. At the moment, it is not clear to me how the problem could be solved. I have already been in contact with the people concerned, but there has been little feedback even from them.

**Comment by ftrojahn** [comment]
Ok, thanx. Our workaround seems to work only, if [netresearch/t3x-rte_ckeditor_image](https://github.com/netresearch/t3x-rte_ckeditor_image/issues/287) is installed and configured.

Couldn't this eventually make sense: have a link field in the RTE dialog box for inserting the iconpack icon? So inserting the link markup would be completely handled here and not by the core ... 

![14_dialog-box-with-link](https://github.com/user-attachments/assets/0c4da241-e01b-4ab6-9beb-4d5b9e19539d)

Or is this completely "unthinkable" ;)

**Comment by SicorDev** [comment]
Hi, 

I got some grey hairs debugging this back and forth, but I found a solution.
First I thought it's maybe just an evil hack, but by now I think it might as well be the official way to do this, not sure...  :-)

I found that deep in the core of CK Editor it is checking whether the Element 'iconpackWebfont' has the attribute 'linkHref'.

So I'm just giving it this attribute in the init method of
EXT:iconpack\Resources\Public\JavaScript\v12\CKEditor\plugin.js

this.attributes = ['linkHref', 'data-iconfig', 'name', 'id', 'class', 'style', 'alt', 'title'], this._defineSchema(), this._defineWebfontConverters(), this._defineImageConverters(), this.editor.commands.add('iconpack', new IconpackCommand(this.editor))

Didn't test it too much and only hacked it straight into a reformatted plugin.js but it works for me to get Facebook and Instagram Icons linked in my project.

Hope this helps.

**Comment by SicorDev** [comment]
Still one odd thing with my solution: It works, unless you separate the icons with a single space. It works either with no space, two or more spaces or enter. if the 2 a Tags are separated *only* by one space (gets converted to nbsp; in sourcecode) the entire content disappears... Weird, but I can live with that for the moment.

**Comment by stephankellermayr** [comment]
Thand for diving into this! I had tried something like this before, but it only worked partially. My main problem was that the icon could then be linked independently of the text. I haven't looked at your solution in detail, but is that the case for you?

**Comment by SicorDev** [comment]
I just tried "Katze Icon Maus" and it wrapped a single link over it:
`<p>
    <a href="https://www.facebook.com/">Katze&nbsp;<span class="fas fa-angles-up fa-2x" data-iconfig="fa6:solid,angles-up,size:2x" linkhref="https://www.facebook.com/"></span> Maus</a>
</p>`

Looks correct to me.

**Comment by stephankellermayr** [comment]
@SicorDev I checked this again and unfortunately came to the same conclusion as in my tests: The icon is given an additional _href_ attribute, which causes the problem that it can be edited as a separate link.

If you select the icon instead of the linked text when editing the link, this creates two different links.

However, as I don't currently have a better solution for this, I will implement it this way for the time being.

The task or problem remains unresolved, but at least the icon can be linked somehow.

Not ideal.

**Comment by stephankellermayr** [comment]
Solved with [cb079b4]
